poetry/puzzle/solver.py

```diff
@@ -21,14 +21,15 @@
 
 
 class Solver:
-    def __init__(self, package, pool, installed, locked, io):
+    def __init__(self, package, pool, installed, locked, io, remove_untracked=False):
```

Comment:
A very minor change needed as we make a concentrated effort to clean up the Poetry codebase by adding type annotations whenever we touch a piece of code. Thank you! 🙌 

```suggestion
    def __init__(
        self,
        package,  # type: ProjectPackage
        pool,  # type: Pool
        installed,  # type: Repository
        locked,  # type: Repository
        io,  # type: ConsoleIO
        remove_untracked=False  # type: bool
    ):
```

Reply:
Thanks. I really like type annotations.

Reply:
Is there a reason behind adding them in the comments rather than directly?